src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -1366,6 +1398,29 @@ after_credit( void *             _ctx,
         init_after_snapshot( ctx );
 
         publish_stake_weights( ctx, mux_ctx, ctx->slot_ctx );
+
+        if( FD_UNLIKELY( ctx->in_wen_restart ) ) {
+          if( FD_UNLIKELY( strncmp( ctx->snapshot, "wksp:", 5 )!=0 ) ) {
+            FD_LOG_WARNING(( "Snapshot should be a funk checkpoint file for wen-restart" ));
```

Comment:
Why cant it be a snapshot file?

Reply:
Let me explain more in the printing here. Below is an explanation.

The work flow has 2 steps: (1) operator terminates FD gracefully; (2) operator restarts FD in wen-restart mode. In step (2), FD should restore the funk/blockstore workspaces of step (1), so FD should load a funk snapshot file during step (2).